src/libopensc/pkcs15-algo.c

```diff
@@ -448,13 +448,18 @@ static struct sc_asn1_pkcs15_algorithm_info algorithm_table[] = {
 #endif
 #ifdef SC_ALGORITHM_EDDSA
 	/* aka Ed25519 */
-	/* RFC 8410, needed to parse/create X509 certs/pubkeys */
-	{ SC_ALGORITHM_EDDSA, {{1, 3, 101, 112, -1}}, NULL, NULL, NULL },
+	/* RFC 8410, needed to parse/create X509 certs/pubkeys  */
+	/* TODO DEE add asn1_decode_ec_params, asn1_encode_ec_params,asn1_free_ec_params */
+	{ SC_ALGORITHM_EDDSA, {{1, 3, 101, 112, -1}}, NULL, NULL, NULL}, /* Ed25119 */
+	{ SC_ALGORITHM_EDDSA, {{1, 3, 6, 1, 4, 11591, 15, 1, -1}}, NULL, NULL, NULL}, /* OpenPGP */
```

Comment:
I think this line comment should have also reference if it is ed25519 (I suspect) or Ed448.

And you miss one "1" in the  OID. I thing you wanted to use the following one:

https://oid-rep.orange-labs.fr/get/1.3.6.1.4.1.11591.15.1

```suggestion
	{ SC_ALGORITHM_EDDSA, {{1, 3, 6, 1, 4, 1, 11591, 15, 1, -1}}, NULL, NULL, NULL}, /* OpenPGP's ed25519 */
```